enterprise/backend/src/metabase_enterprise/database_replication/api.clj

```diff
@@ -28,28 +36,131 @@
 (defn- database-id->connection-id [conns database-id]
   (->> database-id kw-id (get conns) :connection-id))
 
+(defn- schema-filter->fn [{:keys [type patterns]}]
+  (fn [table-schema]
+    (or (= type "all")
+        (try
+          (-> patterns
+              driver.s/schema-pattern->re-pattern
+              (re-matches table-schema)
+              ((case type
```

Comment:
i think i would hoist this and bind it above the `->` threading macro. It takes a second to unpack what's going on there. Feel free to ignore.